src/main/java/seedu/eduke8/parser/MenuParser.java

```diff
@@ -26,34 +30,21 @@ public Command parseCommand(DisplayableList topicList, String userInput) {
         Ui ui = new Ui();
         switch (commandArr[0]) {
         case "about":
-            ui.printAbout();
-            break;
+            return new AboutCommand();
         case "help":
-            ui.printHelp();
-            break;
+            return new HelpCommand();
         case "topics":
-            ((TopicList) topicList).showTopics();
-            break;
+            return new TopicsCommand();
         case "textbook":
-            ui.printTextbook();
-            break;
+            return new TextbookCommand();
         case "quiz":
-            parseQuizCommand((TopicList) topicList, numOfQuestions, topicName, ui);
-            break;
+            return new QuizCommand((TopicList) topicList, numOfQuestions, topicName, ui);
         case "exit":
-            System.exit(0);
-            break;
+            return new ExitCommand();
         default:
             break;
```

Comment:
Maybe we could add something here for users who had wrote commands not in our list. I can create something in Ui